### PR TITLE
feat: add new props to pushprovider

### PR DESCRIPTION
**Pull request by peterdeme**
Add 3 new props

### CODE REVIEW

File: app.go [hunk 1]
```diff
@@ -277,12 +277,15 @@ type PushProvider struct {
...
-	APNKeyID   string `json:"apn_key_id,omitempty"`
-	APNTeamID  string `json:"apn_team_id,omitempty"`
-	APNTopic   string `json:"apn_topic,omitempty"`
-
-	FirebaseCredentials string `json:"firebase_credentials,omitempty"`
+	APNAuthKey              string `json:"apn_auth_key,omitempty"`
+	APNKeyID                string `json:"apn_key_id,omitempty"`
+	APNTeamID               string `json:"apn_team_id,omitempty"`
+	APNTopic                string `json:"apn_topic,omitempty"`
+	APNNotificationTemplate string `json:"apn_notification_template,omitempty"`
```
Comment (ferhatelmas) [inline app.go:284]: we don't want that

Comment (ferhatelmas) [reply app.go:291]: we also need to accept empty string so let's make templates pointer

Comment (peterdeme) [reply app.go:291]: Which ones? 😄 

image

Comment (ferhatelmas) [reply app.go:291]: add:
* `apn_template` to `update_app`
* `firebase_apn_template` to `push_provider`
* `firebase_notification_template` to `push_provider`

File: app.go [hunk 2]
```diff
@@ -282,7 +285,9 @@ type PushProvider struct {
 	APNTeamID  string `json:"apn_team_id,omitempty"`
 	APNTopic   string `json:"apn_topic,omitempty"`
 
-	FirebaseCredentials string `json:"firebase_credentials,omitempty"`
+	FirebaseCredentials          string  `json:"firebase_credentials,omitempty"`
+	FirebaseNotificationTemplate string  `json:"firebase_notification_template,omitempty"`
```
Comment (driver-devel) [inline app.go:289]: We should be able to reset this template to default one the same way as FirebaseAPNTemplate

**Review by driver-devel — APPROVED** [review]
LGTM

**Comment by peterdeme** [comment]
Does this look good @ferhatelmas ?

File: app.go [hunk 3]
```diff
@@ -97,17 +97,19 @@ type APNConfig struct {
...
 
 type FirebaseConfigRequest struct {
-	ServerKey            string `json:"server_key"`
-	NotificationTemplate string `json:"notification_template,omitempty"`
-	DataTemplate         string `json:"data_template,omitempty"`
-	CredentialsJSON      string `json:"credentials_json,omitempty"`
+	ServerKey            string  `json:"server_key"`
+	NotificationTemplate string  `json:"notification_template,omitempty"`
+	DataTemplate         string  `json:"data_template,omitempty"`
+	APNTemplate          *string `json:"apn_template"`
```
Comment (ferhatelmas) [inline app.go:111]: omitempty

File: app.go [hunk 4]
```diff
@@ -135,7 +137,8 @@ type HuaweiConfig struct {
 }
 
 type PushConfigRequest struct {
-	Version string `json:"version"`
+	Version     string `json:"version"`
+	OfflineOnly bool   `json:"offline_only"`
```
Comment (ferhatelmas) [inline app.go:141]: omitempty

**Review by ferhatelmas — APPROVED** [review]
LGTM

**Review by ferhatelmas — APPROVED** [review]

**Pull request merged**